status-models/src/main/java/com/telecominfraproject/wlan/status/equipment/report/models/UnmanagedNeighbourEquipmentInfo.java

```diff
@@ -71,21 +72,24 @@ else if (lowestMac != null && lowestMac.getAddressAsLong() > newMac.getAddressAs
 
     @JsonIgnore
     public int getAverageSignalStrenght(RadioType radioType) {
-        int runningTotal = 0;
-        int numEntries = 0;
-
+        List<Integer> rssiList = new ArrayList<>();
+        
         for (NeighbourRadioInfo radioInfo : this.radios) {
             if (radioInfo.getRadioType() == radioType) {
                 for (NeighbourBssidInfo bssInfo : radioInfo.getBssIds()) {
-                    runningTotal += bssInfo.getSignal();
-                    numEntries++;
+                    rssiList.add(bssInfo.getRssi());
                 }
             }
-
         }
 
-        if (numEntries > 0) {
-            return runningTotal / numEntries;
+        if (rssiList.size() > 0) {
```

Comment:
In Java-8, we can do this in a single line:
`int[] rssiArray = rssiList.stream().mapToInt(i -> i).toArray();`